mlflow/types/type_hints.py

```diff
@@ -262,12 +255,56 @@ def field_required(field: type[FIELD_TYPE]) -> bool:
     return field.is_required()
 
 
+def _get_internal_type_of_list_type_hint(type_hint: type[list[Any]]) -> Any:
+    """
+    Get the internal type of list[...] type hint
+    """
+    args = get_args(type_hint)
+    # a valid list[...] type hint must only contain one argument
+    if len(args) == 0:
+        raise MlflowException.invalid_parameter_value(
+            f"List type hint must contain the internal type, got {type_hint}"
+        )
+    elif len(args) > 1:
+        raise MlflowException.invalid_parameter_value(
+            f"List type hint must contain only one internal type, got {type_hint}"
+        )
+    else:
+        return args[0]
+
+
+def _is_list_type_hint(type_hint: type[Any]) -> bool:
+    origin_type = _get_origin_type(type_hint)
+    return origin_type is list
+
+
+def _infer_schema_from_list_type_hint(type_hint: type[list[Any]]) -> Schema:
+    """
+    Infer schema from a list type hint.
+    The type hint must be list[...], and the inferred schema contains a
+    single ColSpec, where the type is based on the internal type of the list type hint,
+    since ColSpec represents a column's data type of the dataset.
+    e.g. list[int] -> Schema([ColSpec(type=DataType.long, required=True)])
+    A valid `predict` function of a pyfunc model must use list type hint for the input.
+    """
+    if not _is_list_type_hint(type_hint):
+        raise InvalidTypeHintException(
+            message=f"Type hint must be list[...] with a valid internal type, got {type_hint}",
+        )
+    internal_type = _get_internal_type_of_list_type_hint(type_hint)
+    return _infer_schema_from_type_hint(internal_type)
+
+
 def _infer_schema_from_type_hint(type_hint: type[Any]) -> Schema:
+    """
+    Infer schema from a type hint.
+    """
     col_spec_type = _infer_colspec_type_from_type_hint(type_hint)
     # Creating Schema with unnamed optional inputs is not supported
     if col_spec_type.required is False:
-        raise MlflowException.invalid_parameter_value(
-            "If you would like to use Optional types, use a Pydantic-based type hint definition."
+        raise InvalidTypeHintException(
+            message="If you would like to use Optional types, "
+            "use a Pydantic-based type hint definition."
```

Comment:
```suggestion
        raise InvalidTypeHintException(
            message="To define Optional inputs, use a Pydantic-based type hint definition."
```
Is there any doc we can refer here for the usage of pydantic-based definition?